docs/contributors/git-workflow.md

```diff
@@ -0,0 +1,19 @@
+# Git Workflow
+
+## Keeping Your Branch Up To Date
+
+When many different people are working on a project simultaneously, pull requests can go stale quickly. A "stale" pull request is one that is no longer up to date with the main line of development, and it needs to be updated before it can be merged into the project.
+
+While it is tempting to merge from `master` into your branch frequently, this leads to a messy history because each merge creates a merge commit. When working by yourself, it is best to use `git pull --rebase master`, but if you're pushing to a shared repo, it is best to not do any merging or rebasing until the feature is ready for testing, and then do a [rebase](https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request) at the very end. This is one reason why it is important to open pull requests whenever you have working code.
```

Comment:
I'm almost tempted to drop this paragraph. It starts by mentioning a process (plain `merge`) which the user should avoid. Then it seems suggests `pull --rebase` but it turns out that that command is not really adequate, and the reason[^1] for this isn't very clear to me.

Can we keep it short and say the following?
- The preferred method for staying current isn't to merge but to rebase.
- Rebasing means rewriting your changes as if they're happening on top of everything else in `master`.
- Work should be shared early on.

[^1]: _but if you're pushing to a shared repo, it is best to not do any merging or rebasing until the feature is ready for testing, and then do a [rebase](https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request) at the very end. This is one reason why it is important to open pull requests whenever you have working code._

Reply:
Also, I think it should be `git pull --rebase origin master`, but I agree that it's not worth mentioning it if not used in the instructions below. For a start, on a forked version of the repo, `origin` would be the fork (e.g. talldan/gutenberg), rather than wordpress/gutenberg. The fork's master branch might be out of date.

Reply:
@mcsf would you mind rephrasing this paragraph in the way you suggested? :D
Yes, I agree that mentioning `git pull --rebase master` is very confusing.

Reply:
I'll give it go.